src/components/stationsSelect.tsx

```diff
@@ -36,6 +38,87 @@ export function StationSelect(props: Props) {
   } = props;
 
   const [open, setOpen] = useState(false);
+  const [dropDownCurrentState, setDropDownCurrentState] = useState(<></>)
+  const [allStationsLoaded, setAllStationsLoaded] = useState(false)
+  
+  //loading all stations into the dropdown (takes longer so only done if the user clicks on the "load all stations" button)
+  function loadAllStations() {
+    if(!availableStations) return
+    setAllStationsLoaded(true)
+    setDropDownCurrentState(
+      <CommandGroup>
+        {availableStations.map((station) => (
+          <CommandItem
+            key={station.id}
+            value={station.name}
+            onSelect={(currentValue) => {
+              const selectedStation = availableStations.find(
+                (station) =>
+                  station.name.toLowerCase() === currentValue,
+              );
+
+              setOpen(false);
+              setAllStationsLoaded(false)
+              if (selectedStation) onSelect(selectedStation);
+            }}
+          >
+            <Check
+              className={cn(
+                "mr-2 h-4 w-4",
+                selectedStations.find(
+                  (selected) => selected.id == station.id,
+                )
+                  ? "opacity-100"
+                  : "opacity-0",
+              )}
+            />
+            {station.name}
+          </CommandItem>
+        ))}
+      </CommandGroup>
+    )
+  }
+
+  //runs everytime the dropdown is opened
+  React.useEffect(() => {
+    if(!availableStations) return
+    setAllStationsLoaded(false)
+    //setting the dropdown to the first 800 stations to achieve a faster opening of the dropdown
+    setDropDownCurrentState(
+      <CommandGroup>
+        {/*the .filter function here limits the station array to the first 800*/}
+        {availableStations.filter((item, idx) => idx < 800).map((station) => (
```

Comment:
its more performant to use .slice instead of .filter for that

Reply:
also can we make the 800 a constant?

Reply:
but .slice removes everything that comes after the 800 stations so we would have to query the API again. .filter doesn't do that. And yes I can make the 800 a constant :)

Reply:
No `availableStations.slice(0, 800).map(...` is the exact same as `availableStations.slice(0, 800).map(...` but less performant

Reply:
Oh I read `.splice` instead of `.slice` my bad